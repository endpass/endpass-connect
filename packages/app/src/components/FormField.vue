<template>
  <label class="form-field">
    <p v-if="label" class="form-field__label">
      {{ label }}
    </p>
    <div class="form-field__content">
      <slot />      
    </div>
  </label>
</template>

<script>
  export default {
    name: 'FormField',

    props: {
      label: {
        type: String,
        default: null
      }
    }
  }
</script>

<style lang="postcss">
  .form-field {
    display: block;
    margin-bottom: 20px;
  }

  .form-field__label {
    margin-bottom: 10px;
    font-size: 14px;
  }
</style>