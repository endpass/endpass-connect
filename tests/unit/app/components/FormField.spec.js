import { shallowMount } from '@vue/test-utils';
import FormField from '@@/app/src/components/FormField.vue';

describe('FormField', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallowMount(FormField);
    });

    it('should correctly render FormField component without label by default', () => {
      expect(wrapper.name()).toBe('FormField');
      expect(wrapper.find('.form-field__label').exists()).toBe(false);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render label if it passed', () => {
      wrapper = shallowMount(FormField, {
        propsData: {
          label: 'foo',
        },
      });

      expect(wrapper.find('.form-field__label').text()).toBe('foo');
    });
  });
});
