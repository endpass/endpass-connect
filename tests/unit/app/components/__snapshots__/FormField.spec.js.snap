// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField render should correctly render FormField component without label by default 1`] = `
<label class="form-field">
  <!---->
  <div class="form-field__content"></div>
</label>
`;
